tlsutil/config.go

```diff
@@ -375,20 +375,17 @@ func (c *Configurator) check(config Config, pool *x509.CertPool, cert *tls.Certi
 
 	// Ensure we have a CA and cert if VerifyIncoming is set
 	if config.anyVerifyIncoming() {
-		autoEncryptMsg := " AutoEncrypt only secures the connection between client and server and doesn't affect incoming connections on the client."
 		if pool == nil {
-			errMsg := "VerifyIncoming set, and no CA certificate provided!"
-			if config.AutoEncryptTLS {
-				errMsg += autoEncryptMsg
-			}
-			return fmt.Errorf(errMsg)
+			// both auto-config and auto-encrypt while being able to provide the agents certificate and some ca certs require
+			// verifying the connection to the server. Therefore its an error if we ever want to verify incoming certificates
+			// but have no CA certs
+			return fmt.Errorf("VerifyIncoming set but no CA certificates were provided")
 		}
-		if cert == nil {
-			errMsg := "VerifyIncoming set, and no Cert/Key pair provided!"
-			if config.AutoEncryptTLS {
-				errMsg += autoEncryptMsg
-			}
-			return fmt.Errorf(errMsg)
+
+		// We will use the auto_encrypt/auto_config cert for TLS in the incoming APIs when available. Therefore the check
+		// here will ensure that either we enabled one of those two features or a certificate and key were provided manually
+		if cert == nil && !config.AutoEncryptTLS {
```

Comment:
You don't check for auto_config here. I think with your latest addition auto_config is independent from auto_encrypt.

Reply:
Apparently setting `AutoEncryptTLS` wasn't necessary for auto config to work. I could still push the TLS certificates in and have them be used.

Regardless I refactored things a bit to rename things from `AutoEncryptTLS` to just `AutoTLS` with similar naming updates to the other Update* functions specific to the AutoTLS certificates.

Also I updated the runtime.go function for generating the TLS config to enable AutoTLS if either auto config or auto encrypt are enabled.